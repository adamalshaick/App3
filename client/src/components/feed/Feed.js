import React from "react";
import Profiles from "../profiles/Profiles";
import { Header } from "../common/styles/Header";
import Posts from "../posts/Posts";
import fetchProfile from "../common/hoc/fetchProfile";
import PropTypes from "prop-types";

export const Feed = ({ profile }) => {
  return (
    <div className="container entry">
      <div className="row">
        <div
          style={{
            borderLeft: "whitesmoke solid 2px",
            borderRight: "whitesmoke solid 2px"
          }}
          className="col-md-6 mt-5"
        >
          <Header className="text-center">Profiles</Header>
          <hr />
          <Profiles currentProfile={profile.currentProfile} />
        </div>
        <div
          style={{ borderRight: "whitesmoke solid 2px" }}
          className="col-md-6 mt-5"
        >
          <Header className="text-center">Posts</Header>
          <hr />
          <Posts currentProfile={profile.currentProfile} />
        </div>
      </div>
    </div>
  );
};

Feed.propTypes = {
  profile: PropTypes.object.isRequired
};

export default fetchProfile(Feed);
