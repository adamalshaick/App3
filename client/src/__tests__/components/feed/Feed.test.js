import React from "react";
import { shallow } from "enzyme";
import { Feed } from "../../../components/feed/Feed";
import toJson from "enzyme-to-json";

describe("render component", () => {
  const Props = {
    profile: {
      currentProfile: {}
    }
  };

  const wrapper = shallow(<Feed {...Props} />);
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
