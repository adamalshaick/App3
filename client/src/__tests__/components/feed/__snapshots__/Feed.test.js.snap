// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="container entry"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 mt-5"
      style={
        Object {
          "borderLeft": "whitesmoke solid 2px",
          "borderRight": "whitesmoke solid 2px",
        }
      }
    >
      <styled.header
        className="text-center"
      >
        Profiles
      </styled.header>
      <hr />
      <Connect(ComposedComponent)
        currentProfile={Object {}}
      />
    </div>
    <div
      className="col-md-6 mt-5"
      style={
        Object {
          "borderRight": "whitesmoke solid 2px",
        }
      }
    >
      <styled.header
        className="text-center"
      >
        Posts
      </styled.header>
      <hr />
      <Connect(ComposedComponent)
        currentProfile={Object {}}
      />
    </div>
  </div>
</div>
`;
